### PR TITLE
#745 #587 Provide smoke tests with new user creation feature

### PR DESCRIPTION
**Pull request by ivaniuminov**
For each smoke test run, new sandbox user is created (instead of using anton.brueckner and max.musterman). This pull request covers issues:
https://github.com/adorsys/open-banking-gateway/issues/587
https://github.com/adorsys/open-banking-gateway/issues/745

### CODE REVIEW

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java [hunk 1]
```diff
@@ -55,4 +66,53 @@ public SELF enabled_redirect_sandbox_mode(String aspspProfileUri) {
 
         return self();
     }
+
+    public SELF create_new_user(String tppServerUrl, String login, String password) {
+        String body = readResource("restrecord/tpp-ui-input/params/new-user-registration-body.json")
+                              .replace(USER_NAME_PLACEHOLDER, login)
+                              .replace(PASSWORD_PLACEHOLDER, password)
+                              .replace(EMAIL_PLACEHOLDER, login + "@gmail.com");
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java:74]: Please avoid using real domains like `gmail.com`, use `example.com` instead

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java [hunk 2]
```diff
@@ -55,4 +66,53 @@ public SELF enabled_redirect_sandbox_mode(String aspspProfileUri) {
 
         return self();
     }
+
+    public SELF create_new_user(String tppServerUrl, String login, String password) {
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java:70]: Please rename `create_new_user` -> `create_new_user_in_sandbox_tpp_management`
Please rename `tppServerUrl` to sandboxTppManagementServerUrl

TPP is actually OpenBanking, so it is confusing

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java [hunk 3]
```diff
@@ -55,4 +66,53 @@ public SELF enabled_redirect_sandbox_mode(String aspspProfileUri) {
...
+                    .body(body)
+                .when()
+                    .post(tppServerUrl + "/tpp/users")
+                .then()
+                    .statusCode(HttpStatus.CREATED.value());
+
+        return self();
+    }
+
+    public SELF create_account_for_user(String tppServerUrl, String user) {
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java:88]: Please rename `create_account_for_user` -> `create_account_for_user_in_sandbox_tpp_management`
Please rename `tppServerUrl` to sandboxTppManagementServerUrl

TPP is actually OpenBanking, so it is confusing

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java [hunk 4]
```diff
@@ -55,4 +66,53 @@ public SELF enabled_redirect_sandbox_mode(String aspspProfileUri) {
...
+                    .body(body)
+                .when()
+                    .post(tppServerUrl + "/tpp/accounts")
+                .then()
+                    .statusCode(HttpStatus.CREATED.value());
+
+        return self();
+    }
+
+    public SELF deposit_user_account(String tppServerUrl, String user) {
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java:104]: Please rename `deposit_user_account` -> `deposit_to_sandbox_tpp_management_user_account`
Please rename `tppServerUrl` to sandboxTppManagementServerUrl

TPP is actually OpenBanking, so it is confusing

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java [hunk 5]
```diff
@@ -119,6 +119,21 @@ public SELF fintech_calls_list_transactions_for_anton_brueckner(String resourceI
         return self();
     }
 
+    public SELF fintech_calls_list_transactions_for_user(String resourceId, String user) {
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:122]: Please follow
```
given()
    .action()
.when()
     .anotherAction()
```
formatting pattern

File: smoke-tests/src/test/java/de/adorsys/opba/smoketests/OpbaApiWithConsentUiSmokeE2ETest.java [hunk 6]
```diff
@@ -45,6 +45,8 @@
 
     private final String opbaLogin = UUID.randomUUID().toString();
     private final String opbaPassword = UUID.randomUUID().toString();
+    private final String userLogin = UUID.randomUUID().toString();
```
Comment (valb3r) [inline smoke-tests/src/test/java/de/adorsys/opba/smoketests/OpbaApiWithConsentUiSmokeE2ETest.java:48]: please rename 
userLogin -> sandboxUserLogin
userPassword -> sandboxUserPasswod

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java [hunk 7]
```diff
@@ -204,6 +204,42 @@ public SELF open_banking_reads_anton_brueckner_transactions_using_consent_bound_
...
+            String user, String resourceId, LocalDate dateFrom, LocalDate dateTo, String bookingStatus
+    ) {
+        ExtractableResponse<Response> response = getTransactionListFor(user, resourceId, dateFrom, dateTo, bookingStatus);
+
+        this.responseContent = response.body().asString();
+        DocumentContext body = JsonPath.parse(responseContent);
+
+        assertThat(body).extracting(it -> it.read("$.transactions.booked[*].creditorAccount.iban")).asList()
+                .containsOnly(
+                        "DE97500105179747157812"
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java:218]: Creating random user with the same IBAN as a possibly existing user (like test ran twice so two user attempt to exist with the same IBAN) - won't it cause constraint violations on Sandbox side? (as IBAN should be unique)

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java (hunk 1 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java (hunk 3 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java]: renamed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java (hunk 2 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java]: renamed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java (hunk 4 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-sandbox/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/sandbox/servers/SandboxServers.java]: renamed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java (hunk 5 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:122]: done

File: smoke-tests/src/test/java/de/adorsys/opba/smoketests/OpbaApiWithConsentUiSmokeE2ETest.java (hunk 6 above)
Comment (ivaniuminov) [reply smoke-tests/src/test/java/de/adorsys/opba/smoketests/OpbaApiWithConsentUiSmokeE2ETest.java]: renamed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java (hunk 7 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java:218]: random IBAN generation added

**Pull request merged**